vault/resource_secrets_sync_association.go

```diff
@@ -0,0 +1,182 @@
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Type of secrets store.",
+				ForceNew:    true,
+			},
+			consts.FieldMount: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Specifies the mount where the secret is located.",
+			},
+			fieldAccessor: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "Specifies the unique accessor of the mount.",
+				ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+			},
+			fieldSecretName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Specifies the name of the secret to synchronize.",
+			},
+			fieldSyncStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Specifies the status of the association.",
+			},
+			fieldUpdatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Duration string stating when the secret was last updated.",
+			},
+		},
+	}
+}
+
+func secretsSyncAssociationWrite(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	name := d.Get(consts.FieldName).(string)
+	destType := d.Get(consts.FieldType).(string)
+	path := secretsSyncAssociationSetPath(name, destType)
+
+	data := map[string]interface{}{}
+
+	for _, k := range []string{
+		fieldSecretName,
+		consts.FieldMount,
+	} {
+		data[k] = d.Get(k)
+	}
+
+	log.Printf("[DEBUG] Writing association to %q", path)
+	resp, err := client.Logical().WriteWithContext(ctx, path, data)
+	if err != nil {
+		return diag.Errorf("error setting secrets sync association %q: %s", path, err)
+	}
+	log.Printf("[DEBUG] Wrote association to %q", path)
+
+	// expect accessor to be provided from mount
+	accessor := d.Get(fieldAccessor).(string)
+
+	// associations can't be read from Vault
+	// set data that is received from Vault upon writes
+	vaultRespKey := fmt.Sprintf("%s/%s", accessor, name)
+	associatedSecrets := "associated_secrets"
+	secretData, ok := resp.Data[associatedSecrets]
+	if !ok {
+		// did not find any associated secrets for this mount
+		// we expect to see data here since we just wrote an association
+		return diag.Errorf("expected associated secrets; received no secret associations in mount")
+	}
+
+	for _, k := range secretsSyncAssociationFields {
+		if secretsMap, ok := secretData.(map[string]interface{}); ok {
+			if val, ok := secretsMap[vaultRespKey]; ok {
+				if valMap, ok := val.(map[string]interface{}); ok {
+					if v, ok := valMap[k]; ok {
+						if err := d.Set(k, v); err != nil {
+							return diag.FromErr(err)
+						}
+					}
+				}
+			}
+		}
+	}
+
+	d.SetId(path)
+
+	return nil
+}
+
+func secretsSyncAssociationRead(_ context.Context, _ *schema.ResourceData, _ interface{}) diag.Diagnostics {
+	return nil
+}
```

Comment:
This could be implemented with the request mentioned above. Until all association fields cause a recreate it might not be necessary? Maybe help the TFVP recover more gracefully if an association is deleted out-of-band?

Reply:
Gotcha, thanks for the context! I can investigate adding this in and the implications on the TF state, and depending on that it can be part of a follow-up PR 👍🏼